pkg/component/controller/cplb_unix.go

```diff
@@ -274,31 +268,25 @@ func (*Keepalived) getLinkAddresses(link netlink.Link) ([]netlink.Addr, []string
 }
 
 func (k *Keepalived) generateKeepalivedTemplate() error {
-	f, err := os.OpenFile(k.K0sVars.KeepalivedConfigFile, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, fs.FileMode(0500))
-	if err != nil {
-		return fmt.Errorf("failed to open keepalived config file: %w", err)
-	}
-	defer f.Close()
-
-	template, err := template.New("keepalived").Parse(keepalivedConfigTemplate)
-	if err != nil {
-		return fmt.Errorf("failed to parse keepalived template: %w", err)
-	}
-
+	template := template.Must(template.New("keepalived").Parse(keepalivedConfigTemplate))
```

Comment:
```suggestion
	template, err := template.New("keepalived").Parse(keepalivedConfigTemplate)
	if err != nil {
		return fmt.Errorf("parse keepalived template: %w", err)
	}
```

Reply:
I think those "Must" style functions are specifically meant for const inputs like here. Why bother with error handling if the input is guaranteed to be valid? Using `template.Must(...)` here is no different from using e.g. `regex.MustParse("[a-z0-9]")`.

Reply:
I think it's to ease defining globals:

```go
var parsedTemplate = template.Must(template.New("keepalived").Parse(keepalivedConfigTemplate))

func main() {
   parsedTemplate.Execute(...)
}
```

At least, that seems to be the intention of `regexp.MustCompile`:

> MustCompile is like [Compile] but panics if the expression cannot be parsed.
> It simplifies safe initialization of global variables holding compiled regular
> expressions.

So, as the template is constant, it could be parsed at startup:

```diff
-const keepalivedConfigTemplate = `
+var keepalivedConfigTemplate = template.Must(template.New("keepalived").Parse(`
```